go/kbfs/test/edit_history_test.go

```diff
@@ -602,3 +602,54 @@ func TestEditHistoryRenameDirAndReuseNameForLink(t *testing.T) {
 		),
 	)
 }
+
+// Regression test for HOTPOT-803.
+func TestEditHistoryUnflushedRenameOverNewFile(t *testing.T) {
+	// Alice creates two files in the first revision, but then creates
+	// 2 files in the second revision that are renamed over the
+	// original two files.  She also creates two files that are removed.
+	expectedEdits := []expectedEdit{
+		{
+			"alice",
+			keybase1.FolderType_PRIVATE,
+			"alice",
+			[]string{
+				"/keybase/private/alice/a",
+				"/keybase/private/alice/b",
+			},
+			[]string{
+				"/keybase/private/alice/f",
+				"/keybase/private/alice/e",
+			},
+		},
+	}
+
+	test(t, journal(),
+		users("alice"),
+		as(alice,
+			mkfile("a", "foo a"),
+			mkfile("b", "foo d"),
+		),
+		as(alice,
+			enableJournal(),
+		),
+		as(alice,
+			pwriteBSSync("c", []byte("c foo"), 0, false),
+			pwriteBSSync("d", []byte("d foo"), 0, false),
+			rename("b", "e"),
+			pwriteBSSync("f", []byte("f foo"), 0, false),
+			rename("a", "f"),
+			rename("c", "b"),
+			pwriteBSSync("a", []byte("g foo"), 0, false),
+			rename("d", "a"),
+			rm("e"),
+			rm("f"),
```

Comment:
It seems to me that this test case is more complicated than it needs to be to exercise this flow, unless I'm missing something.

Reply:
@jzila yeah it's probably doing twice the work -- I just literally transcribed the behavior I saw in the log into a test (changing the filenames to protect the innocent).  Do you think it's worth simplifying it?  Can do so, would just have to back out the fix so I can test whether it still fails in the same way when simplified.

Reply:
Maybe keep this one as a regression for the specific scenario that the user hit, and also have a simplified case for the scenario you fixed.

Reply:
Cool done, thanks!